main.cpp

```diff
@@ -1,475 +1,581 @@
+    float aveGrade = 3.45f;
+
+    //train teachers
+    void teacherTraining(bool internalTraining, std::string trainingRequired);
+    //students christmas play
+    void christmasPlay(int numberOfStudentsInCast, std::string nameOfPlay, int rehearsalDays);
+    //close for holidays
+    void holidayClosure(int howManyDaysClosed, std::string nameOfHoliday);
+};
+
 /*
-Thing 5)
+Thing 3) Farm
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of sheep (int)
+    2) number of cows (int)
+    3) crop yeild (double)
+    4) number of farmers (int)
+    5) area in hectare (double)
 3 things it can do:
-    1)
-    2)
-    3)
- */
+    1) sell sheep
+    2) harvest crop
+    3) construct new barn
+*/
+
+struct Farm
+{
+    //number of sheep (int)
+    int numSheep = 1234;
+    //number of cows (int)
+    int numCows = 321;
+    //crop yeild (double)
+    double cropYield = 76543.21;
+    //number of farmers (int)
+    int numFarmers = 6;
+    //area in hectare (double)
+    double farmLandAreaInHectare = 84.3;
+
+    //sell sheep
+    float getTurnoverSheepSale(float priceOfSheep, int numSheep); //returns the income from the sale of this farms sheep
+    //harvest crop
+    void harvestCrop(bool isCropMature, int numFarmHands, bool machinesFueled);
+    //construct new barn
+    void constructNewBarn(bool needNewBarn, float newConstructionBudget);
+};
 
 /*
-Thing 6)
+Thing 4) Commerical Aeroplane
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of engines (int)
+    2) top speed (double)
+    3) number of seats (int)
+    4) fuel tank capacity (double)
+    5) flight range (double)
 3 things it can do:
-    1)
-    2)
-    3)
- */
+    1) take off
+    2) emergency landing
+    3) inflight entertainment
+*/
+
+struct CommericalAeroplane
+{
+    //number of engines (int)
+    int numEngines = 4;
+    //top speed (double)
+    double maxAirSpeedMPH = 576.43;
+    //number of seats (int)
+    int numSeats = 524;
+    //fuel tank capacity (double)
+    double fuelTankCapacityInLitres = 238840.04;
+    //flight range (double)
+    double flightRangeInKM = 14310.23;
+
+    struct Cockpit
+    {
+        float age = 1.35f;
+        int numSeats = 5;
+        int numWindows = 4;
+        bool hasBlackBox = true;
+        bool ElectronicFlightInstrumentSystem = true;
+
+        void increaseAirSpeed(float deltaV, float currentAirSpeed);
+        void groundProximityWarningAlarm(bool nearGround, float proximityWarningThreshold);
+        void autopilotDisengagement(bool pilotAndCopilotChecklistComplete);
+    };
+
+
+    //take off
+    bool takeOffComplete(float groundSpeed, bool landingGear, bool armDoorsAndCrossCheck); //returns whether has this instance of Commercial Areoplane completed a successful take off
+    //emergency landing
+    void emergencyLanding(bool enginefailure, float urrentAirSpeed);
+    //inflight entertainment
+    void inflightEntertainment(int passengerSeatNumber, std::string movieSelction);
```

Comment:
Your function names do not match the plain-english descriptions.
And some of your plain-english descriptions are not actions.
They should match almost 1:1
- [ ] come up with better plain-english actions.  Use a verb in the function names.
- [ ] revise